File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        res = float("inf")
        l = 0
        curr = 0
        for r in range(len(blocks)):
            if r - l + 1 < k:
                curr += int(blocks[r] == "W")
                continue
            curr += int(blocks[r] == "W")
            res = min(res, curr)
            curr -= int(blocks[l] == "W")
            l += 1
        return res

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.go
func minimumRecolors(blocks string, k int) int {
    res := math.MaxInt32
    var (curr, l int) 
    for r:=0; r < len(blocks); r++{
        if (r - l + 1 < k){
            if blocks[r] == 'W'{
                curr ++;
            }
            continue
        }
        if blocks[r] == 'W'{
            curr ++;
        }
        res = min(res, curr)
            
        if blocks[l] == 'W'{
            curr --;
        }
        l++

    }
    return res
}